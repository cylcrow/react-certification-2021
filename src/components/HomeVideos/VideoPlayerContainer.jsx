import React, { useEffect } from 'react';
import styled from 'styled-components';

const StyledVideoPlayerContainer = styled.div``;

const VideoPlayerContainer = () => {
  useEffect(() => {
    let done = false;
    /* global YT */
    /* eslint no-undef: "error" */
    const player = new YT.Player('player', {
      height: '390',
      width: '640',
      videoId: 'M7lc1UVf-VE',
      events: {
        onReady(event) {
          event.target.playVideo();
        },
        onStateChange(event) {
          if (event.data === YT.PlayerState.PLAYING && !done) {
            setTimeout(player.stopVideo, 6000);
            done = true;
          }
        },
      },
    });
  });

  return (
    <StyledVideoPlayerContainer>
      <div id="player" />
    </StyledVideoPlayerContainer>
  );
};

export default VideoPlayerContainer;
