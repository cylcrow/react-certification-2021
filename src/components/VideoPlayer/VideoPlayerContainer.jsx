import React, { useContext, useEffect } from 'react';
import styled from 'styled-components';
import SearchContext from '../../providers/SearchContext';
import RelatedVideoList from './RelatedVideoList';

const StyledVideoPlayerContainer = styled.div`
  display: flex;
`;

const VideoPlayerContainer = ({ videoId }) => {
  const { videos } = useContext(SearchContext);

  useEffect(() => {
    let done = false;
    /* global YT */
    /* eslint no-undef: "error" */
    window.player = new YT.Player('player', {
      height:
        0.5 *
        Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0),
      width: '60%',
      videoId,
      events: {
        onReady(event) {
          event.target.playVideo();
        },
        onStateChange(event) {
          if (event.data === YT.PlayerState.PLAYING && !done) {
            setTimeout(window.player.stopVideo, 6000);
            done = true;
          }
        },
      },
    });
  });

  return (
    <StyledVideoPlayerContainer id="video-player-container">
      <div id="player" />
      <RelatedVideoList
        videos={videos}
        playVideoByID={(videoID) => window.player.loadVideoById(videoID, 0)}
      />
    </StyledVideoPlayerContainer>
  );
};

export default VideoPlayerContainer;
