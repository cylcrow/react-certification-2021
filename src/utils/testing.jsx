import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}