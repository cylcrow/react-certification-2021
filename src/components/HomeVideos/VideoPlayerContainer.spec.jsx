import React from 'react';
import 'jest-styled-components';
import { renderWithTheme, YTMockedObject } from '../../utils';
import VideoPlayerContainer from './VideoPlayerContainer';

global.YT = YTMockedObject;

const build = (Component = <VideoPlayerContainer />) => {
  const { container } = renderWithTheme(Component);
  return { container };
};

describe('VideoPlayerContainer', () => {
  it('shows div for iframe load', () => {
    const { firstChild } = build().container;
    expect(firstChild).toMatchSnapshot();
  });
});
