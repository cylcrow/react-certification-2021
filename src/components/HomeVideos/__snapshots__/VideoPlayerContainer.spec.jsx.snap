// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayerContainer shows div for iframe load 1`] = `
<div
  class="sc-bdVaJa fePDys"
>
  <div
    id="player"
  />
</div>
`;
